### ISSUE TITLE
Smf-elements-nn.txt files are formatted as Unix files (EOL LF) not Windows (EOL CRLF).

### ISSUE THREAD

**Issue by Bruce-95** [ClimberMel/SMF_Add-in#74]
smf-Elements-nn.txt files listed [here ](https://climbermel.github.io/SMF_Add-in/Elements )and in the latest Add-in zip file  are formatted with LF (line-feed) as the end-of-line character. This is causing the elements to be read into arrays incorrectly. VBA requires CRLF when using VBA's Line Input# statement. Users are reporting multiple errors when using RCHGetElementNumber(). Problem first reported on May 15, 2024 [here](https://groups.google.com/g/smf-addin/c/1ApjOtrFW-8/m/vjoPo2QACQAJ).

**Comment by Bruce-95** [comment]
Opening issue for tracking.

Should investigate which options to use in the .gitattributes file as suggested [here](https://groups.google.com/g/smf-addin/c/1ApjOtrFW-8/m/7Bqub1LlBwAJ).

**Comment by DecimalTurn** [comment]
Are the files available from the website showing the raw files from GitHub like it is available from [this URL](https://raw.githubusercontent.com/ClimberMel/SMF_Add-in/main/docs/Elements/smf-elements-0.txt) (for examaple) or is there some kind of `git checkout` process that occurs?

**Comment by ClimberMel** [comment]
There should be a process created on github to update the files for the website.  Another thing for the to do list...Sent from my Galaxy
-------- Original message --------From: Martin Leduc ***@***.***> Date: 2024-07-08  2:42 p.m.  (GMT-08:00) To: ClimberMel/SMF_Add-in ***@***.***> Cc: Subscribed ***@***.***> Subject: Re: [ClimberMel/SMF_Add-in] Smf-elements-nn.txt files are formatted as Unix files (EOL LF) not Windows (EOL CRLF). (Issue #74) 
Are the files available from the website showing the raw files from GitHub like it is available from this URL (for examaple) or is there some kind of git checkout process that occurs?

**Comment by DecimalTurn** [comment]
Ah, so this is a manual process. Then, I'm assuming it's tempting to just download the raw file from the GitHub repo to update the website, but the way this repo is currently set up means that the LF version is the one stored as the "raw file" for GitHub.

![image](https://github.com/ClimberMel/SMF_Add-in/assets/31558169/3fc6991c-f7bd-4e17-9b16-f8b3f8a09929)

This is due to the fact that `* text=auto` will cause all files detected as text to be converted to LF when committed to the index and GitHub uses the file in the index as the "raw file". 

Unfortunately, the modification in https://github.com/ClimberMel/SMF_Add-in/pull/85 won't affect the raw file since the proposed changes only influence the checkout process. However, merging the PR would indeed fix the "download ZIP" version since GitHub does perform the equivalent of a `git checkout` when creating that .zip file.

![image](https://github.com/ClimberMel/SMF_Add-in/assets/31558169/b5c5d868-764e-44fe-a9c0-d7bedf15471b)

**Comment by ClimberMel** [comment]
Yes, too much of it is a manual process.  I kept meaning to get more into the processing, but when it was just me there didn't seem like a lot of advantage to process.Sent from my Galaxy
-------- Original message --------From: Martin Leduc ***@***.***> Date: 2024-07-08  7:31 p.m.  (GMT-08:00) To: ClimberMel/SMF_Add-in ***@***.***> Cc: ClimberMel ***@***.***>, Comment ***@***.***> Subject: Re: [ClimberMel/SMF_Add-in] Smf-elements-nn.txt files are formatted as Unix files (EOL LF) not Windows (EOL CRLF). (Issue #74) 
Ah, so this is a manual process. Then, I'm assuming it's tempting to just download the raw file from the GitHub repo to update the website, but the way this repo is currently set up means that the LF version is the one stored as the "raw file" for GitHub.
image.png (view on web)
This is due to the fact that * text=auto will cause all files detected as text to be converted to LF when committed to the index and GitHub uses the file in the index as the "raw file".
Unfortunately, the modification in #85 won't affect the raw file since the proposed changes only influence the checkout process. However, merging the PR would indeed fix the "download ZIP" version since GitHub does perform the equivalent of a git checkout when creating that .zip file.
image.png (view on web)

**Comment by Bruce-95** [comment]
This is from the [MS documentation](https://learn.microsoft.com/en-us/office/vba/language/reference/user-interface-help/line-inputstatement?source=recommendations) for VBA's Line Input# statement -

"The Line Input # statement reads from a file one character at a time until it encounters a carriage return (Chr(13)) or carriage return-linefeed (Chr(13) + Chr(10)) sequence. Carriage return-linefeed sequences are skipped rather than appended to the character string."

Coming up on 2 years and still not fixed>
